src/AppInstallerCommonCore/Public/winget/Resources.h

```diff
@@ -55,5 +62,36 @@ namespace AppInstaller
         // Resource data is valid as long as the binary is loaded.
         std::pair<const BYTE*, size_t> GetResourceAsBytes(int resourceName, int resourceType);
         std::pair<const BYTE*, size_t> GetResourceAsBytes(PCWSTR resourceName, PCWSTR resourceType);
+
+        struct ResourceOpenException : std::exception
+        {
+            ResourceOpenException(const winrt::hresult_error& hre);
+
+            const char* what() const noexcept override { return m_message.c_str(); }
+
+        private:
+            std::string m_message;
+        };
+
+        // TODO https://task.ms/40934960 Consider renaming or removing LocString

```

Comment:
@lechacon, `LocString` was responsible for taking `StringId` and resolve it from the `.resw` file. Now that we can produce `LocIndString` directly using the added `StringId::operator()` operator, we can directly store resolved strings in `LocIndString` without using `LocString`, and for `StringId` instances (the case of localized strings with no placeholders) we can add a constructor in `LocIndString` which simply calls the `()` with no parameters (which what `LocString` does in this PR).

@JohnMcPMS, removing `LocString` struct will result in a lot of changes across many files which I wanted to avoid in this PR. But I created a task to better evaluate this change along with any unexpected compilations issues.